mlflow/pyfunc/utils.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+import mlflow.pyfunc
+from mlflow.utils import PYTHON_VERSION, get_major_minor_py_version
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.tracking.utils import _get_model_log_dir
+
+
+def _get_code_dirs(src_code_path, dst_code_path=None):
+    """
+    Obtains the names of the subdirectories contained under the specified source code
+    path and joins them with the specified destination code path.
+
+    :param src_code_path: The path of the source code directory for which to list subdirectories.
+    :param dst_code_path: The destination directory path to which subdirectory names should be
+                          joined.
+    """
+    if not dst_code_path:
+        dst_code_path = src_code_path
+    return [(os.path.join(dst_code_path, x))
+            for x in os.listdir(src_code_path) if not x.endswith(".py") and not
+            x.endswith(".pyc") and not x == "__pycache__"]
+
+
+def _warn_potentially_incompatible_py_version_if_necessary(model_py_version=None):
+    """
+    Compares the version of Python that was used to save a given model with the version
+    of Python that is currently running. If a major or minor version difference is detected,
+    logs an appropriate warning.
+    """
+    if model_py_version is None:
+        mlflow.pyfunc._logger.warning(
```

Comment:
we should probably  not use pyfunc._logger here?

Reply:
We want to, so I've moved this method back to `pyfunc/__init__.py` and removed the `utils` module since it should really only contain a single method: `_get_code_dirs`.